extended-tracer/src/test/java/io/opentelemetry/contrib/tracer/TracingTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.tracer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatException;
+import static org.junit.jupiter.api.Named.named;
+
+import com.google.errorprone.annotations.Keep;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.sdk.testing.junit5.OpenTelemetryExtension;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class TracingTest {
+
+  interface ThrowingBiConsumer<T, U> {
+    void accept(T t, U u) throws Throwable;
+  }
+
+  @RegisterExtension
+  static final OpenTelemetryExtension otelTesting = OpenTelemetryExtension.create();
+
+  private final Tracing tracing = new Tracing(otelTesting.getOpenTelemetry(), "test");
+
+  @Test
+  void propagation() {
+    tracing.run(
+        "parent",
+        () -> {
+          Map<String, String> propagationHeaders = tracing.getPropagationHeaders();
+          assertThat(propagationHeaders).hasSize(1).containsKey("traceparent");
+
+          tracing.traceServerSpan(propagationHeaders, tracing.spanBuilder("child"), () -> null);
+        });
+
+    otelTesting
+        .assertTraces()
+        .hasSize(1)
+        .hasTracesSatisfyingExactly(
+            trace ->
+                trace.hasSpansSatisfyingExactly(
+                    SpanDataAssert::hasNoParent, span -> span.hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void callWithBaggage() {
+    String value =
+        tracing.call(
+            "parent",
+            () ->
+                tracing.callWithBaggage(
+                    Collections.singletonMap("key", "value"),
+                    () -> Baggage.current().getEntryValue("key")));
+
+    assertThat(value).isEqualTo("value");
+  }
+
```

Comment:
it could be worth adding tests for other public methods, not only for the coverage but for showing usage examples